com.unity.ml-agents/Tests/Editor/ModelRunnerTest.cs

```diff
@@ -0,0 +1,100 @@
+using NUnit.Framework;
+using UnityEngine;
+using UnityEditor;
+using Barracuda;
+using MLAgents.InferenceBrain;
+using MLAgents.Sensors;
+using System.Linq;
+
+namespace MLAgents.Tests
+{
+    [TestFixture]
+    public class ModelRunnerTest
+    {
+        const string k_continuous2vis8vec2actionPath = "Packages/com.unity.ml-agents/Tests/Editor/Resources/continuous2vis8vec2action.nn";
+        const string k_discrete1vis0vec_2_3action_recurrModelPath = "Packages/com.unity.ml-agents/Tests/Editor/Resources/discrete1vis0vec_2_3action_recurr.nn";
+        NNModel continuous2vis8vec2actionModel;
+        NNModel discrete1vis0vec_2_3action_recurrModel;
+        Test3DSensorComponent sensor_21_20_3;
+        Test3DSensorComponent sensor_20_22_3;
+
+        private BrainParameters GetContinuous2vis8vec2actionBrainParameters()
+        {
+            var validBrainParameters = new BrainParameters();
+            validBrainParameters.vectorObservationSize = 8;
+            validBrainParameters.vectorActionSize = new int[] { 2 };
+            validBrainParameters.numStackedVectorObservations = 1;
+            validBrainParameters.vectorActionSpaceType = SpaceType.Continuous;
+            return validBrainParameters;
+        }
+
+        private BrainParameters GetDiscrete1vis0vec_2_3action_recurrModelBrainParameters()
+        {
+            var validBrainParameters = new BrainParameters();
+            validBrainParameters.vectorObservationSize = 0;
+            validBrainParameters.vectorActionSize = new int[] { 2, 3 };
+            validBrainParameters.numStackedVectorObservations = 1;
+            validBrainParameters.vectorActionSpaceType = SpaceType.Discrete;
+            return validBrainParameters;
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            continuous2vis8vec2actionModel = (NNModel)AssetDatabase.LoadAssetAtPath(k_continuous2vis8vec2actionPath, typeof(NNModel));
+            discrete1vis0vec_2_3action_recurrModel = (NNModel)AssetDatabase.LoadAssetAtPath(k_discrete1vis0vec_2_3action_recurrModelPath, typeof(NNModel));
+            var go = new GameObject("SensorA");
+            sensor_21_20_3 = go.AddComponent<Test3DSensorComponent>();
+            sensor_21_20_3.Sensor = new Test3DSensor("SensorA", 21, 20, 3);
+            sensor_20_22_3 = go.AddComponent<Test3DSensorComponent>();
+            sensor_20_22_3.Sensor = new Test3DSensor("SensorA", 20, 22, 3);
```

Comment:
```suggestion
            sensor_20_22_3.Sensor = new Test3DSensor("SensorB", 20, 22, 3);
```

Reply:
I'm surprised that doesn't cause an error, but maybe the Agent never gets activated.